giving-example/public/adyenGiving.js

```diff
@@ -0,0 +1,86 @@
+const clientKey = document.getElementById("clientKey").innerHTML;
+
+async function callServer(url, data) {
+  const res = await fetch(url, {
+    method: "POST",
+    body: data ? JSON.stringify(data) : "",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+
+  return await res.json();
+}
+
+async function handleDonation(donationToken, pspReference, amount) {
+  try {
+    const res = await callServer(`/api/donations?donationToken=${encodeURIComponent(donationToken)}&pspReference=${pspReference}`, amount);
+
+    switch (res.status) {
+      case "completed":
+        window.location.href = "/result/donated";
+        break;
+      default:
+        window.location.href = "/result/error";
+        break;
+    }
+  } catch (error) {
+    console.error(error);
+    alert("Error occurred. Look at console for details");
+  }
+
+
+}
+
+async function startGiving() {
+
+  const checkout= await AdyenCheckout(
+    {
+      clientKey,
+      environment: "test",
+    }
+  );
+
+  const donationConfig = {
+    amounts: {
+      currency: "EUR",
+      values: [300, 500, 1000]
+    },
+    // backgroundUrl: "https://www.adyen.com/dam/jcr:38701562-9572-4aae-acd3-ed53e220d63b/social-responsibility-city-illustration.svg",
+    description: "The Charitable Foundation is a non-profit aiming at showing you the power of Adyen Giving",
+    logoUrl: "https://www.adyen.com/dam/jcr:49277359-f3b5-4ceb-b54c-08189ae2433e/hands-rockon-icon-green.svg",
+    name: "The Charitable Foundation",
+    url: "https://www.adyen.com/social-responsibility/giving",
+    showCancelButton: true,
+    disclaimerMessage: {
+      message: "By donating you agree to the %#terms%#",
+      linkText: "terms and conditions",
+      link: "https://www.adyen.com/legal/terms-and-conditions" // Replace with yours
+
+
+    },
+    onDonate: (state, component) => {
+      if(state.isValid) {
+        console.log("Initiating donation");
+        let donationToken = sessionStorage.getItem("donationToken");
```

Comment:
We store the plain-text donationToken on the frontend, what are the best practices here. Wondering whether we could have the backend store this value (and do a look-up based on the PspReference)

Reply:
yes, I think we need to refactor this (and change it in the Java sample too. I will rework it.

Reply:
Refactored to store donation token in session